pkg/webhook/nodenetworkconfigurationpolicy/validation.go

```diff
@@ -0,0 +1,50 @@
+package nodenetworkconfigurationpolicy
+
+import (
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	shared "github.com/nmstate/kubernetes-nmstate/api/shared"
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/api/v1beta1"
+)
+
+func onPolicySpecChange(policy nmstatev1beta1.NodeNetworkConfigurationPolicy, currentPolicy nmstatev1beta1.NodeNetworkConfigurationPolicy) bool {
+	return !reflect.DeepEqual(policy.Spec, currentPolicy.Spec)
+}
+
+func validateSequentialPolicyNotInProgressHook(policy nmstatev1beta1.NodeNetworkConfigurationPolicy, currentPolicy nmstatev1beta1.NodeNetworkConfigurationPolicy) []metav1.StatusCause {
+	causes := []metav1.StatusCause{}
+	if currentPolicy.Spec.Parallel && policy.Spec.Parallel {
+		return causes
+	}
+
+	currentPolicyDegradedCondition := currentPolicy.Status.Conditions.Find(shared.NodeNetworkConfigurationPolicyConditionDegraded)
+	currentPolicyAvailableCondition := currentPolicy.Status.Conditions.Find(shared.NodeNetworkConfigurationPolicyConditionAvailable)
+
+	if currentPolicyAvailableCondition == nil ||
+		currentPolicyDegradedCondition == nil ||
+		(currentPolicyDegradedCondition.Status == corev1.ConditionUnknown && currentPolicyAvailableCondition.Status == corev1.ConditionUnknown) {
```

Comment:
To check that a policy is progressing checking the Reason is enough, you can take a look at the code listing that set progressing state [1]

[1] https://github.com/nmstate/kubernetes-nmstate/blob/master/pkg/policyconditions/conditions.go#L25-L39

Reply:
Right, thanks!